rancher2/structure_cluster_aks_config.go

```diff
@@ -135,8 +156,8 @@ func flattenClusterAKSConfig(in *AzureKubernetesServiceConfig) ([]interface{}, e
 		obj["subscription_id"] = in.SubscriptionID
 	}
 
-	if len(in.Tag) > 0 {
-		obj["tag"] = toMapInterface(in.Tag)
+	if len(in.Tags) > 0 {
+		obj["tags"] = toMapInterface(in.Tags)
 	}
 
```

Comment:
@rawmind0 thanks for the review.
In my testing tag didn’t work because it arrives at rancher/kontainer-engine via the providers flatten method which means it’s flattened to tag and k-e expects the plural - tags.
Therefore “supporting” both would stop the terraform from throwing an error; but still wouldn’t work unless k-e is also updated to support incoming “tag” attribute.
My initial intended approach was to have the provider simply translate tag->tags on flattening; but i don’t think the current architecture will support that type of translation (unless you can maybe suggest otherwise - you will know this code better than I)

Reply:
> In my testing tag didn’t work because it arrives at rancher/kontainer-engine via the providers flatten method which means it’s flattened to tag and k-e expects the plural - tags.

Yeah, it's flattened to `Tag` on golang struct, but the golang object is marshalled as json format before make the Rancher API call. It's defined as `tags` on json format, `json:"tags,omitempty" yaml:"tags,omitempty"`

> Therefore “supporting” both would stop the terraform from throwing an error; but still wouldn’t work unless k-e is also updated to support incoming “tag” attribute.
My initial intended approach was to have the provider simply translate tag->tags on flattening; but i don’t think the current architecture will support that type of translation (unless you can maybe suggest otherwise - you will know this code better than I)

I'm suggesting to use `Tags` on golang struct (as you proposed), but map it to `tag` and `tags` terraform arguments for a while. Flatten and expand would translate golang struct `Tags` to `tag` and `tags` tf arguments and the reverse.